src/scripts/sync_synonym_curation_filtering.py

```diff
@@ -0,0 +1,148 @@
+"""Filter out cases where curation is needed"""
+import os
+import shutil
+import sys
+from argparse import ArgumentParser
+from datetime import datetime
+from pathlib import Path
+from typing import Union, List
+
+import pandas as pd
+from oaklib import get_adapter
+from oaklib.types import CURIE, URI
+
+from src.scripts.utils import remove_angle_brackets
+
+HERE = Path(os.path.abspath(os.path.dirname(__file__)))
+SRC_DIR = HERE.parent
+PROJECT_ROOT = SRC_DIR.parent
+sys.path.insert(0, str(PROJECT_ROOT))
+from src.scripts.sync_synonym import _common_operations, _read_sparql_output_tsv
+
+
+def _read_synonym_file(path: Union[Path, str], case: str) -> pd.DataFrame:
+    """Does special operations for reading in a synonym sync robot.tsv in this context"""
+    df = pd.read_csv(path, sep='\t').rename(columns={'synonym_scope_source': 'synonym_scope'}) \
+        .drop(0)[['synonym', 'synonym_scope', 'mondo_id', 'source_id', 'synonym_type']].fillna('')
+    df['synonym_type'] = df['synonym_type'].apply(
+        lambda x: x.replace('http://purl.obolibrary.org/obo/mondo#ABBREVIATION', 'MONDO:ABBREVIATION'))
+    df['case'] = case
+    return df
+
+
+def sync_synonyms_curation_filtering(
+    added_path: Union[Path, str], confirmed_path: Union[Path, str], updated_path: Union[Path, str],
+    mondo_synonyms_path: Union[Path, str], mondo_db_path: Union[Path, str], outpath: Union[Path, str]
+):
+    """Filter out cases where curation is needed"""
+    # todo temp: backup temporarily for development; remove when done
```

Comment:
Is this `todo` comment still relevant or should be removed now?

Reply:
Hmm, good question. I'll remove the todo as it will be obsoleted this change I have which is next on my todo list:
- Make it so that the first round of outputs from the synonym sync go to `tmp/`. Then use those as the inputs for this script, and the outputs of it go into `reports/`. This eliminates the need for a backup, as the files will no longer be mutated. I'll remove the todo.

FYI that change doesn't have to hold up this PR. I can do it after.

Reply:
Yeah, the original idea was to mutate these files. The new idea came up later, but ideally I should have gotten to it by now.

It sounds like you are ready to approve this PR? I just wanna ask because I see no other remaining threads, but there is no approval yet.